modules/zivid/point_cloud.py

```diff
@@ -0,0 +1,66 @@
+"""Contains the PointCloud class."""
+import numpy
+
+import _zivid
+
+
+class PointCloud:
+    """A point cloud."""
+
+    def __init__(self, internal_point_cloud):
+        """Create a point cloud from an internal point cloud.
+
+        Args:
+            internal_point_cloud: a internal point cloud
+
+        """
+        if not isinstance(internal_point_cloud, _zivid.PointCloud):
+            raise ValueError(
+                "Unsupported type for argument internal_point_cloud. Got {}, expected {}".format(
+                    type(internal_point_cloud), type(_zivid.PointCloud)
+                )
+            )
+        self.__impl = internal_point_cloud
+
+    def to_array(self):
+        """Convert point cloud to numpy array.
+
+        Returns:
+            a numpy array
+
+        """
+        self.__impl.assert_not_released()
+        return numpy.array(self.__impl)
+
+    @property
+    def height(self):
+        """Return height (number of rows) of point cloud.
+
+        Returns:
+            a positive integer
+
+        """
+        return self.__impl.height()
+
+    @property
+    def width(self):
+        """Return width (number of columns) of point cloud.
+
+        Returns:
+            a positive integer
+
+        """
+        return self.__impl.width()
+
+    def release(self):
+        """Release the underlying resources."""
+        self.__impl.release()
+
+    def __enter__(self):
```

Comment:
Maybe we should have a mixin or base class or something for these three functions. They are duplicated many places now.

You are just adding to the existing solution, so no need to fix in this PR. But if you agree please add an issue for it.

Reply:
Could probably be done through a decorator:
```
@utils.context_manager
class SomethingThatRequiresContextManager:
    ....
```
Will create an issue

Reply:
https://github.com/zivid/zivid-python/issues/27